presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -875,7 +891,13 @@ public void dropTable(String databaseName, String tableName, boolean deleteData)
                     .stopOnIllegalExceptions()
                     .run("dropTable", stats.getDropTable().wrap(() -> {
                         try (ThriftMetastoreClient client = createMetastoreClient()) {
+                            Table table = client.getTable(databaseName, tableName);
                             client.dropTable(databaseName, tableName, deleteData);
+                            String tableLocation = table.getSd().getLocation();
+                            if (deleteData && isManagedTable(table) && !isNullOrEmpty(tableLocation)) {
+                                // Workaround for issue https://github.com/prestosql/presto/issues/1053
```

Comment:
Let's remove this comment. The code for `deleteTableFilesIfNeeded` makes it obvious that we are deleting files when the path is on S3.